examples/ASDKgram/Sample/AppDelegate.m

```
@@ -0,0 +1,116 @@
+  
+  self.window.rootViewController = tabBarController;
+  [self.window makeKeyAndVisible];
+  
+  // opaque status bar background
+  self.statusBarOpaqueUnderlayView                 = [[UIView alloc] init];
+  self.statusBarOpaqueUnderlayView.backgroundColor = [UIColor darkBlueColor];
+  self.statusBarOpaqueUnderlayView.frame           = [[UIApplication sharedApplication] statusBarFrame];
+  if ( ![[UIApplication sharedApplication] isStatusBarHidden] ) {
+    [[[UIApplication sharedApplication] keyWindow] addSubview:self.statusBarOpaqueUnderlayView];
+  }
+  [[NSNotificationCenter defaultCenter] addObserver:self
+                                           selector:@selector(handleDidChangeStatusBarOrientationNotification:)
+                                               name:UIApplicationWillChangeStatusBarOrientationNotification
+                                             object:nil];
+  return YES;
+}
+
+// UITabBarControllerDelegate
+- (void)tabBarController:(UITabBarController *)tabBarController didSelectViewController:(UIViewController *)viewController;
+{
+  if ([viewController isKindOfClass:[UINavigationController class]]) { 
+    NSArray *viewControllers = [(UINavigationController *)viewController viewControllers];
+    UIViewController *rootViewController = viewControllers[0];
+    if ([rootViewController conformsToProtocol:@protocol(PhotoFeedViewControllerProtocol)]) {
+      // FIXME: the dataModel does not currently handle clearing data during loading properly
+//      [(id <PhotoFeedViewControllerProtocol>)rootViewController resetAllData];
+    }
+  }
+}
+
+- (void)handleDidChangeStatusBarOrientationNotification:(NSNotification *)notification;
+{
+  NSNumber *newOrientation = [notification.userInfo objectForKey: UIApplicationStatusBarOrientationUserInfoKey];
+  NSInteger newOrientationInt = [newOrientation integerValue];
+  if (UIInterfaceOrientationIsLandscape(newOrientationInt)) {
+    [self.statusBarOpaqueUnderlayView removeFromSuperview];
```

looks like iOS8 defaults to hiding the status bar in landscape orientation. I moved the OpaqueUnderlay to a UIWindow subclass for re-layout ease.